src/tests/core/blocks/test_handle_block.c

```diff
@@ -0,0 +1,219 @@
+#include <stdlib.h>
+#include "minunit.h"
+#include "handle_block.h"
+#include "create_block.h"
+#include "mempool.h"
+#include "utxo_pool.h"
+#include "blockchain.h"
+#include "wallet_pool.h"
+#include "utxo_to_tx.h"
+#include "crypto.h"
+
+int tests_run = 0;
+
+mbedtls_ecdsa_context *last_key_pair;
+Transaction *_make_tx() {
+  Transaction *tx;
+  Input *in;
+  Output *out;
+  mbedtls_ecdsa_context *key_pair;
+  mbedtls_ecdsa_context *output_key_pair;
+  tx = malloc(sizeof(Transaction));
+  in = malloc(sizeof(Input));
+  out = malloc(sizeof(Output));
+
+  memset(in, 0, sizeof(Input));
+  key_pair = gen_keys();
+  in->pub_key = &(key_pair->private_Q);
+  in->sig_len = 0;
+
+  memset(out, 0, sizeof(Output));
+
+  output_key_pair = gen_keys();
+  hash_pub_key(out->public_key_hash, output_key_pair);
+  last_key_pair = output_key_pair;
+  tx->num_inputs = 1;
+  tx->inputs = in;
+  tx->num_outputs = 1;
+  tx->outputs = out;
+
+  return tx;
+}
+
+void _fill_mempool(){
+  Transaction *input_tx, *tx1;
+  input_tx = _make_tx();
+  input_tx->outputs[0].amt = 100;
+  utxo_pool_init();
+  utxo_pool_add(input_tx, 0);
+
+  mbedtls_ecdsa_context *input_tx_context = malloc(sizeof(mbedtls_ecdsa_context));
+  memcpy(input_tx_context, last_key_pair, sizeof(mbedtls_ecdsa_context));
+
+  tx1 = _make_tx();
+  hash_tx(tx1->inputs[0].prev_tx_id, input_tx);
+  tx1->inputs[0].prev_utxo_output = 0;
+  tx1->inputs[0].pub_key = &input_tx_context->private_Q;
+  tx1->outputs[0].amt = 90;
+
+  unsigned char temp_hash[TX_HASH_LEN];
+  hash_tx(temp_hash, tx1);
+  tx1->inputs[0].sig_len = write_sig(tx1->inputs[0].signature, SIGNATURE_LEN, temp_hash, TX_HASH_LEN, input_tx_context);
+  mempool_init();
+  mempool_add(tx1);
```

Comment:
This will make merge errors in main, my PR #40 takes care of utxo addition withing mempool add so this will cause errors.

Note there is actually a weird thing now. UTXO to TX should probably be in the mempool cmake library. and we should force utxo_to_tx init to happen within mempool init since they need to be always tied. THoughts?

Reply:
Agree on both counts, will do